packages/mui-material/src/SnackbarContent/SnackbarContent.d.ts

```diff
@@ -9,7 +9,7 @@ export interface SnackbarContentProps extends StandardProps<PaperProps, 'childre
   /**
    * The action to display. It renders after the message, at the end of the snackbar.
    */
-  action?: React.ReactNode;
+  action?: React.ReactNode | ((key: string) => React.ReactNode);
```

Comment:
I don't follow this change. `<SnackbarContent>` doesn't seem to support this.

```suggestion
  action?: React.ReactNode;
```

Reply:
It's added to support callback type in `action` method. See: 
```jsx
<SnackbarsProvider
  ref={snackbarRef}
  autoHideDuration={4000}
  action={(key) => (
    <React.Fragment>
      <Button
        color="secondary"
        size="small"
        onClick={snackbarRef.current!.close(key)}
      >
        UNDO
      </Button>
      <IconButton
        size="small"
        aria-label="close"
        color="inherit"
        onClick={snackbarRef.current!.close(key)}
      >
        <CloseIcon fontSize="small" />
      </IconButton>
    </React.Fragment>
  )}
>
  <MyApp />
</SnackbarsProvider>;
```

Reply:
Isolated action prop in https://github.com/mui/material-ui/pull/31991/commits/0e7d8978b6d904cfab6a2d27e9a9b2557b9c7b9c.